### PR TITLE
Report zero values instead of unknown for empty ingest queries

### PR DESCRIPTION
**Pull request by abhishekrb19**
MSQ now allows empty ingest queries by default. For such queries that don't generate any output rows, the query counters in the async status result object/task report don't contain `numTotalRows` and `totalSizeInBytes`. These properties when not set/undefined can be confusing to API clients. For example, the web-console treats it as unknown values.

This patch fixes the counters by explicitly reporting them as 0 instead of null for empty ingest queries.

- Add a test case each for an empty insert and empty replace query in `SqlMSQStatementResourcePostTest.java`. Validate the responses which now include `ResultSetInformation` with 0 rows instead of null.
- Add test coverage for all the MSQ Destinations, including `DataSourceMSQDestination` and `TaskReportMSQDestination` in `SqlStatementResourceHelperTest.java`. We don't report 0 values for `DurableStorageDestination` type as it only applies to select queries as it can be unknown, so the behavior is preserved as-is.
- Some miscellaneous spacing, typo and interpolation log fixes in `QueryHostFinder.java` that I found while testing this patch.

Before the fix, web-console interpreting as unknown number of rows:

CleanShot 2024-01-11 at 23 39 53@2x

After the fix:

CleanShot 2024-01-11 at 23 39 15@2x

<hr>

##### Key changed/added classes in this PR
 * SqlStatementResourceHelper.java
 * SqlStatementResourceHelperTest.java

<hr>

This PR has:

- [x] been self-reviewed.
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/SqlStatementResourceHelper.java [hunk 1]
```diff
@@ -221,16 +216,24 @@ private static Optional<List<PageInformation>> populatePagesForDurableStorageDes
       throw DruidException.defensive("Expected worker count to be set for stage[%d]", finalStage);
     }
 
-
     List<PageInformation> pages = new ArrayList<>();
+
```
Comment (cryptoe) [inline extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/SqlStatementResourceHelper.java:220]: nit:Extra spaces.

File: extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/SqlStatementResourceHelper.java [hunk 2]
```diff
@@ -221,16 +216,24 @@ private static Optional<List<PageInformation>> populatePagesForDurableStorageDes
...
+
+    Set<Integer> emptyWorkerCounters = new HashSet<>();
+
     for (int partitionNumber = 0; partitionNumber < totalPartitions; partitionNumber++) {
       for (int workerNumber = 0; workerNumber < totalWorkerCount; workerNumber++) {
         CounterSnapshots workerCounter = workerCounters.get(workerNumber);
+        if ((workerCounter == null || workerCounter.isEmpty()) && !emptyWorkerCounters.contains(workerNumber)) {
+          pages.add(new PageInformation(pages.size(), 0L, 0L, workerNumber, null));
+          emptyWorkerCounters.add(workerNumber);
```
Comment (cryptoe) [inline extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/SqlStatementResourceHelper.java:229]: I donot understand why we require a hashSet here. 
Also, DurableStorageDestination is only for selects and not inserts. Not sure if changes are required here.

Comment (abhishekrb19) [reply extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/SqlStatementResourceHelper.java]: Got it. Reverted the changes for `DurableStorageDestination` type as we want to report 0 value counters only for ingest queries

**Comment by cryptoe** [comment]
@abhishekrb19 Could you also add a release note for the above PR.

**Pull request merged**